content/articles/how-to-create-a-reusable-react-form/index.md

```diff
@@ -0,0 +1,468 @@
+
+```bash
+`npm install react-router-dom`
+```
+
+Press enter to install it.
+
+Next, we then import BrowserRouter, Routes and Route from react-router-dom. We wrap BrowserRouter in opening and closing tags, within it we wrap Routes also with opening and closing tags and wrap the Route with a self closing tag. We can't render our element in Route without wrapping it with Routes.
+When rendering Route, we need to add two properties called element and path, the element accepts the component we want to render to the DOM and the path accepts the location we want our component to be rendered.
+
+As shown below:
+
+```javascript
+import {Route, BrowserRouter, Routes} from 'react-router-dom';
+
+const App = () => {
+  return (
+    <div className={/*style name*/}>
+      <BrowserRouter>
+        <Routes>
+            <Route/>
+        </Routes>
+      </BrowserRouter>
+    </div>
+  );
+}
+
+export default App;
+```
+Then we import our Auth.js and Contact file into the App.js file and render it within the return bracket. In our browser DOM when we visit `localhost:3000`, we'll see the Contact page. When we visit `localhost:3000/auth`, we'll see the Auth page.
+
+Our App.js file should look like this below:
+
+```javascript
+import {Route, BrowserRouter, Routes} from 'react-router-dom';
+
+const App = () => {
+  return (
+    <div className={/*style name*/}>
+      <BrowserRouter>
+        <Routes>
+            <Route path='/auth' element={<Auth/>}/>
+            <Route path="/" element={<Contact />} />
+        </Routes>
+      </BrowserRouter>
+    </div>
+  );
+}
+
+export default App;
+```
+
+![App](/engineering-education/how-to-create-a-reusable-react-form/app.png)
+
+#### Conclusion
+How to create a Reusable React Form.
+Forms are important in every website or application, we can use them for many things in our application.
+React is all about components, we created a component to handle our HTML codes
+and a useForm hook component to handle our state change.
+In this tutorial, we created a reusable react form which can be used or rendered in any part of our application.
+We used useReducer, which is a React hook that accept two arguments, a reducer and an initialState.
+The reducer contains your state logic, kinda like setState while the initialState is your initial state. useReducer also returns your current state and also a dispatch method which is used to trigger the state logic.
+This ensures we get the values of our input where ever it is rendered.
+By doing this tutorial, we were able to create a reusable form component which can be used in any part of our application or project.
+React make our coding work less stressful, being able to use a component over over is just so amazing, we only need to make sure each of our component can be reusable and we are good to go.
```

Comment:
Make sure your conclusion is brief. Remove that summary section below it. Integrate that info into the conclusion section instead.